app/src/sharedTest/java/org/oppia/android/app/parser/ListItemLeadingMarginSpanTest.kt

```diff
@@ -446,6 +475,235 @@ class ListItemLeadingMarginSpanTest {
+      displayLocale = displayLocale
+    )
+    val (textView, htmlResult) = activityScenarioRule.scenario.runWithActivity {
+      testCoroutineDispatchers.runCurrent()
+      val textView: TextView = it.findViewById(R.id.test_list_content_text_view)
+      val htmlResult = htmlParser.parseOppiaHtml(
+        "<ul>" +
+          "        <li> Usage Data\", such as:" +
+          "          <ul>" +
+          "            <li>your answers to Lessons;</li>" +
+          "            <li>when you begin and end a Lesson;</li>" +
+          "          </ul>" +
+          "        </li>" +
+          "        <li> any contributions you make to the Site (such as feedback on" +
+          "            Lessons, edits to Lessons, and Lessons created);</li>" +
+          "      </ul>",
+        textView
+      )
+      textView.text = htmlResult
+      return@runWithActivity textView to htmlResult
+    }
+    assertThat(displayLocale.getLayoutDirection()).isEqualTo(ViewCompat.LAYOUT_DIRECTION_RTL)
+    assertThat(textView.textDirection).isEqualTo(View.TEXT_DIRECTION_RTL)
+    /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
+    val bulletSpans =
+      htmlResult.getSpans(
+        0,
+        htmlResult.length,
+        ListItemLeadingMarginSpan.UlSpan::class.java
+      )
+    assertThat(bulletSpans.size.toLong()).isEqualTo(4)
+    val bulletSpan0 = bulletSpans[0] as ListItemLeadingMarginSpan.UlSpan
+    assertThat(bulletSpan0).isNotNull()
+    htmlResult.setSpan(
+      bulletSpan0,
+      0,
+      3,
+      Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+    )
+    bulletSpan0.drawLeadingMargin(
+      canvas, paint, x, dir, 0, 0, 96, htmlResult,
+      htmlResult.getSpanStart(bulletSpan0),
+      htmlResult.getSpanEnd(bulletSpan0),
+      true,
+      null
+    )
+    val bulletSpan1 = bulletSpans[1] as ListItemLeadingMarginSpan.UlSpan
+    bulletSpan1.drawLeadingMargin(
+      canvas, paint, x, dir, 96, 0, 182, htmlResult,
+      htmlResult.getSpanStart(bulletSpan1),
+      htmlResult.getSpanEnd(bulletSpan1),
+      true,
+      null
+    )
+
+    val bulletSpan2 = bulletSpans[2] as ListItemLeadingMarginSpan.UlSpan
+    assertThat(bulletSpan2).isNotNull()
+    bulletSpan2.drawLeadingMargin(
+      canvas, paint, x, dir, 182, 0, 268, htmlResult,
+      htmlResult.getSpanStart(bulletSpan2),
+      htmlResult.getSpanEnd(bulletSpan2),
+      true,
+      null
+    )
+    val bulletSpan3 = bulletSpans[3] as ListItemLeadingMarginSpan.UlSpan
+    assertThat(bulletSpan3).isNotNull()
+    bulletSpan3.drawLeadingMargin(
+      canvas, paint, x, dir, 354, 0, 440, htmlResult,
+      htmlResult.getSpanStart(bulletSpan3),
+      htmlResult.getSpanEnd(bulletSpan3),
+      true,
+      null
+    )
+
+    val shadowCanvas = shadowOf(canvas)
+    // The below assertion verifies the 1st item of inner level list
+    assertThat(shadowCanvas.getDrawnCircle(0).centerX).isWithin(1e-5f).of(-73.0f)
```

Comment:
Is negative actually correct? That would result in the bullet being offscreen.

Also, shouldn't these be larger values since they would be drawn on the other side of the canvas (per RTL)?

Reply:
Fixed.